### PR TITLE
Update pre-commits, and add more ruff rules

### PR DESCRIPTION
**Pull request by larsevj**
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -176,6 +176,8 @@ select = [
   "F",   # pyflakes
   "PL",  # pylint
   "NPY", # numpy specific rules
+  "C4", # flake8-comprehensions
+  "ASYNC", # flake8-async
```
Comment (berland) [inline pyproject.toml:180]: Here we should decide on aligning hashes, or switch to ruff/black standard which is two spaces before the hash..

File: src/ert/scheduler/lsf_driver.py [hunk 2]
```diff
@@ -342,12 +341,12 @@ async def kill(self, iens: int) -> None:
                 retry_interval=self._sleep_time_between_cmd_retries,
                 exit_on_msgs=(JOB_ALREADY_FINISHED_BKILL_MSG),
             )
-            subprocess.Popen(
+            await asyncio.create_subprocess_shell(
                 f"sleep {self._sleep_time_between_bkills}; {self._bkill_cmd} -s SIGKILL {job_id}",
                 shell=True,
```
Comment (berland) [inline src/ert/scheduler/lsf_driver.py:346]: This argument is most likely implicit when using `create_subprocess_shell`

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -10,14 +10,14 @@ repos:
...
-    rev: v0.4.8
+    rev: v0.5.0
     hooks:
     - id: ruff
       args: [ --fix ]
     - id: ruff-format
 
 -   repo: https://github.com/pre-commit/mirrors-clang-format
-    rev: v18.1.6
+    rev: v18.1.7
```
Comment (andreas-el) [inline .pre-commit-config.yaml:20]: Did you test using `18.1.8` ?

**Comment by berland** [comment]
The second commit has an effectively void commit message..

**Review by berland — APPROVED** [review]
go for it.

**Pull request merged**